sandberg_christy_ndvi.py

```diff
@@ -0,0 +1,700 @@
+    print('\u2705 The data in your date column is datetime!')
+    single_scene_points += 2
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Ensure the site name is correct
+if student_ndvi_ts_single_site.site.values[0] == 'HARV':
+    print('\u2705 You have the correct site name!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct site name.')
+
+if np.allclose(0.281131628228094, student_ndvi_ts_single_site.mean_ndvi.values[0]):
+    print('\u2705 You have the correct mean NDVI value!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct mean ndvi value.')
+
+print("\n \u27A1 You received {} out of 15 points for creating a dataframe.".format(
+    single_scene_points))
+single_scene_points
+
+
+# ## Task 2:
+# 
+# In the cell below, process all of the landsat scenes. Create a DataFrame that contains the following 
+# information for each scene
+# 
+# 
+# |   | index  | site  | mean_ndvi  | 
+# |---|---|---|---|
+# | Date  |   |   |   |
+# |  2017-01-07  | 0  | SJER  | .4  |  
+# 
+# Be sure to call your dataframe at the end of the cell to ensure autograding works.
+# HINT: FOR THIS STEP, leave any rows containing missing values (`NAN`).
+
+# **The process used for creating a dataframe of a single landsat scene is expanded 
+# with a loop, to collect info on all landsat scenes in a directory.**
+
+# In[8]:
+
+
+# Create dataframe of NDVI including the cleaning data to deal with clouds
+
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Don't forget to set date as the index and make the values of type datetime
+
+# YOUR CODE HERE
+
+# Create glob list of all scene folders at given file location
+scene_dirs = glob(os.path.join('ndvi-automation', 'sites', '*', 
+                               'landsat-crop', '*'))
+
+# Create empty list to store info about each landsat scene.
+full_ndvi_list = []
+
+# Loop through each landsat scene folder
+for scene in scene_dirs:
+    
+    # use 'process_single_scene' function to get the site name, data 
+    # acquisition date, pixel_qa & ndvi_bands list 
+    site_loc, acq_date, pixel_qa, ndvi_bands = process_single_scene (
+        scene, (0,10000))
+
+    # use 'calculate_masked_ndvi' function to get get ndvi with cloud mask
+    masked_ndvi = calculate_masked_ndvi(ndvi_bands, pixel_qa)
+
+    # append 'site', 'date', and 'mean_ndvi' to list
+    full_ndvi_list.append([acq_date, site_loc, masked_ndvi.mean().values])
+
+# convert list to dataframe with date as datetime index
+full_ndvi_df = pd.DataFrame(data=full_ndvi_list, 
+                       columns=['date', 'site', 'mean_ndvi'])
+full_ndvi_df['date'] = pd.to_datetime(full_ndvi_df['date'], yearfirst=True, 
+                                 format='%Y-%m-%d')
+full_ndvi_df = full_ndvi_df.set_index('date')
+
+# call dataframe at the end of the cell for autograding 
+full_ndvi_df
+
```

Comment:
@csandberg303 I think your code is great, clean and only repetitive because the assignment by nature asks you to repeat a process by running the same code twice!